dev/ProgressRing/ProgressRing.cpp

```diff
@@ -147,32 +228,54 @@ void ProgressRing::SetLottieBackgroundColor(winrt::impl::com_ref<AnimatedVisuals
         }
         else
         {
-            // Default color fallback if Background() Brush does not contain SolidColorBrush with Color property.
+             //Default color fallback if Background() Brush does not contain SolidColorBrush with Color property.
             return SharedHelpers::FindInApplicationResources(s_DefaultBackgroundThemeResourceName).as<winrt::SolidColorBrush>().Color();
         }
     }();
 
-    progressRingIndeterminate->GetThemeProperties(compositor).InsertVector4(s_BackgroundName, SharedHelpers::RgbaColor(backgroundColor));
+    if (const auto progressRing = animatedVisualSource.try_as<AnimatedVisuals::ProgressRingIndeterminate>())
+    {
+        progressRing->GetThemeProperties(compositor).InsertVector4(s_BackgroundName, SharedHelpers::RgbaColor(backgroundColor));
+    }
 }
 
-void ProgressRing::ChangeVisualState()
+void ProgressRing::UpdateLottieProgress()
 {
-    if (IsActive())
+    if (auto&& determinatePlayer = m_determinatePlayer.get())
     {
-        winrt::VisualStateManager::GoToState(*this, L"Active", true);
+        const double range = Maximum() - Minimum();
+        const double fromProgress = m_oldValue / range;
+        const double toProgress = Value() / range;
+
+        determinatePlayer.PlayAsync(fromProgress, toProgress, false);
+        m_oldValue = Value();
+    }
+}
 
-        if (auto&& player = m_player.get())
+void ProgressRing::UpdateStates()
+{
+    if (IsActive() && IsIndeterminate())
```

Comment:
>IsActive [](start = 8, length = 8)

nit, store these properties so that you don't need to call them in each if check